GoRogue/MapGeneration/Region.cs

```diff
@@ -72,63 +112,25 @@ public Region(Lines.Algorithm algorithm, IComponentCollection? components, param
         /// <param name="corners">The corners of the region</param>
         public Region(params Point[] corners) : this(corners, Lines.Algorithm.DDA) { }
 
+        /// <summary>
+        /// Returns a new Region using the provided area
+        /// </summary>
+        /// <param name="area">This region's Area</param>
+        /// <param name="components">A component collection for this region</param>
+        public Region(PolygonArea area, IComponentCollection? components = null)
+        {
+            Area = area;
+            GoRogueComponents = components ?? new ComponentCollection();
+            GoRogueComponents.ParentForAddedComponents = this;
+        }
+
         private Region(List<Point> corners, Lines.Algorithm algorithm, IComponentCollection? components)
         {
             Area = new PolygonArea(ref corners, algorithm);
             GoRogueComponents = components ?? new ComponentCollection();
             GoRogueComponents.ParentForAddedComponents = this;
         }
```

Comment:
I think we could possibly implement this in terms of the new public constructor?
 ```suggestion
        private Region(List<Point> corners, Lines.Algorithm algorithm, IComponentCollection? components)
            : this(new PolygonArea(ref corners, algorithm), components)) { }
```